pinot-core/src/test/java/org/apache/pinot/queries/TransformQueriesTest.java

```diff
@@ -0,0 +1,286 @@
+
+      String query = "SELECT AVG(SUB(INT_COL1, INT_COL2)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "-1000.00000");
+
+      query = "SELECT AVG(SUB(LONG_COL1, INT_COL1)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "499000.00000");
+
+      query = "SELECT AVG(SUB(LONG_COL2, LONG_COL1)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "500000.00000");
+
+      query = "SELECT AVG(ADD(INT_COL1, INT_COL2)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "3000.00000");
+
+      query = "SELECT AVG(ADD(INT_COL1, LONG_COL1)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "501000.00000");
+
+      query = "SELECT AVG(ADD(LONG_COL1, LONG_COL2)) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "1500000.00000");
+
+      query = "SELECT AVG(ADD(DIV(INT_COL1, INT_COL2), DIV(LONG_COL1, LONG_COL2))) FROM foo";
+      runAndVerifyInterSegmentQuery(query, "1.00000");
+    } finally {
+      delete();
+    }
+  }
+
+  private List<GenericRow> createDataSet(final int numRows) {
+    final ThreadLocalRandom random = ThreadLocalRandom.current();
+    final List<GenericRow> rows = new ArrayList<>(numRows);
+    Object[] columns;
+
+    // ROW
+    GenericRow row = new GenericRow();
+    columns = new Object[]{"Pinot", 1000, 2000, 500000, 1000000};
+    row.putField(D1, columns[0]);
+    row.putField(M1, columns[1]);
+    row.putField(M2, columns[2]);
+    row.putField(M3, columns[3]);
+    row.putField(M4, columns[4]);
+    row.putField(TIME, random.nextLong(TimeUtils.getValidMinTimeMillis(), TimeUtils.getValidMaxTimeMillis()));
+
+    for (int i = 0; i < numRows; i++) {
+      rows.add(row);
+    }
+    return rows;
+  }
+
+  private void runAndVerifyInnerSegmentQuery(String query, double sum, long count) {
+    AggregationOperator aggregationOperator = getOperatorForQuery(query);
+    IntermediateResultsBlock resultsBlock = aggregationOperator.nextBlock();
+    final List<Object> aggregationResult = resultsBlock.getAggregationResult();
+    AvgPair avgPair = (AvgPair)aggregationResult.get(0);
+    Assert.assertEquals(avgPair.getSum(), sum);
+    Assert.assertEquals(avgPair.getCount(), count);
+  }
+
+  private void runAndVerifyInterSegmentQuery(String query, String serialized) {
+    BrokerResponseNative brokerResponse = getBrokerResponseForQuery(query);
+    List<AggregationResult> aggregationResults = brokerResponse.getAggregationResults();
+    Assert.assertEquals(aggregationResults.size(), 1);
+    Serializable value = aggregationResults.get(0).getValue();
+    Assert.assertEquals(value.toString(), serialized);
+  }
+
+  @Override
+  protected String getFilter() {
+    return "";
+  }
+
+  @Override
+  protected IndexSegment getIndexSegment() {
+    return _indexSegments.get(0);
+  }
+
+  @Override
+  protected List<SegmentDataManager> getSegmentDataManagers() {
+    return _segmentDataManagers;
+  }
+
+  private void createSegment(
+      Schema schema,
```

Comment:
Please reformat the file with the Pinot Style

Reply:
done